src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php

```diff
@@ -431,7 +432,7 @@ protected function getDoctrine()
     /**
      * Get a user from the Security Token Storage.
      *
-     * @return object|null
+     * @return UserInterface|object|null
```

Comment:
what about deprecating returning non UserInterface objects? similar like we did in `Security` util:

https://github.com/symfony/symfony/blob/7b0ca3ab5678232f876a95991b66778f4aaa33e3/src/Symfony/Component/Security/Core/Security.php#L51-L56

then we can aim for consistent accessors: `getUser(): ?UserInterface`, apart from the tokenstorage (low level API).

Reply:
otherwise we just document a subset of `object`, creating a trap for users to not do type-checking. That's the wrong signal IMHO.

Reply:
That would mean that we have to create a dedicated user object for anonymous users which may have other implications. But we should discuss that in a different issue anyway IMO. :)

Reply:
currently we have `ControllerTrait::getUser, Security::getUser, TokenInterface::getUser`, all varying in behavior ... just saying :)

At least the first 2 serve the same purpose IMO.

Reply:
It is best to make a final decision on this request.

Reply:
let's continue the discussion with some real code in #34909